sql/functions/run_maintenance.sql

```diff
@@ -261,14 +262,23 @@ LOOP
             v_current_partition_timestamp = CURRENT_TIMESTAMP;
         END IF;
 
+        -- avoids the need for a functional index using to_timestamp or decoder on control column to quickly find the max
+        v_max_control_expression := CASE
+            WHEN v_row.epoch = 'seconds' THEN format('to_timestamp(max(%I))', v_row.control)
+            WHEN v_row.epoch = 'milliseconds' THEN format('to_timestamp((max(%I)/1000)::float)', v_row.control)
+            WHEN v_row.epoch = 'microseconds' THEN format('to_timestamp((max(%I)/1000000)::float)', v_row.control)
+            WHEN v_row.epoch = 'nanoseconds' THEN format('to_timestamp((max(%I)/1000000000)::float)', v_row.control)
+            WHEN v_row.epoch = 'func' THEN format('%s(max(%I))', v_time_decoder, v_row.control)
+            ELSE format('max(%I)', v_row.control)
+        END;
```

Comment:
This should remove the need for a functional index to find the maximum, provided there is an index on the underlying column. That should be a safe assumption to make since postgres prohibits creating a unique index or primary key on a partitioned table that doesn't contain the partition key.

If this works in all scenarios, it should also be ported to `partition_data_time()`